x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -104,7 +104,7 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
 
   const getAnomaliesMapsLink = async (anomaly: MlAnomaliesTableRecord) => {
     const index = job.datafeed_config.indices[0];
-    const dataViewId = await getDataViewIdFromName(index);
+    const dataViewId = await getDataViewIdFromName(data.dataViews, index);
```

Comment:
As mentioned after James's comment  below - the `indexServiceFactory` in `ml/public/application/util/index_service.ts` (which is meant to replace the utils file `getDataViewIdFromName` is pulled from here) should be used here to get that functionality. It only requires `dataViewsService`, which should be available in this component.

Reply:
Updated in [70ab770](https://github.com/elastic/kibana/pull/176788/commits/70ab770cc5e06985dba0dd7b86123eee21f7a72e).